### PR TITLE
189 fix e2e test

### LINKED ISSUE #189: Fix E2E test
Somehow it skips the final part of the test, in which the death of the player is simulated.

< bug, testing

### PR DESCRIPTION
**Pull request by mpsijm**
Will close #189 

Summary of changes:
- The default save directory is backed up so that the E2E runs with an empty save directory (this prevents accidentally loading corrupted save-files
- Refactored the `clickButton` methods to one method (they were all the same anyway, the button index constants already indicate the scenes)
- The player DIES >:D
- The main test method in E2E is looking totally clean now (you should check it out, it's so clean :D)
- Updated documentation of E2E

Have fun reviewing, shouldn't be a big one :)

<review, testing, bug

### CODE REVIEW

File: src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java [hunk 1]
```diff
@@ -302,36 +329,25 @@ private void keyboard(final KeyCode kc) {
...
-                .getChildrenUnmodifiable();
-        clickOn(buttons.get(buttonNo), MouseButton.PRIMARY);
+    private void playerDies() {
+        Track.getInstance().getPlayer().die();
+        OBSERVABLE.notify(
+                GameObserver.Category.PLAYER,
+                GameObserver.Player.COLLISION,
+                AbstractObstacle.class.getSimpleName());
+        sleep(SLEEP_FACTOR * InternalTicker.NANOS_PER_TICK / InternalTicker.E6);
         letPlayerSurvive();            // Make sure there are no obstacles
```
Comment (thervh70) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: Why should the player be surviving if it is dead?

< review, logic

File: src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java [hunk 2]
```diff
@@ -65,7 +70,7 @@
 
     private static final int MAIN_START = 0;
     private static final int MAIN_SETTINGS = 1;
-//    private static final int MAIN_QUIT = 2;
+    private static final int MAIN_QUIT = 2;
     private static final int MAIN_ACCOUNTS = 3;
     private static final int MAIN_SHOP = 4;
 //    private static final int MAIN_HIGHSCORE = 5;
```
Comment (thervh70) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: Why does the test contain this commented variable?

< layout

File: src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java [hunk 3]
```diff
@@ -364,13 +380,6 @@ private void pausePopup(final int buttonNo) {
...
 
-    private void playerDies() {
-        Track.getInstance().getPlayer().die();
-        sleep(SLEEP_FACTOR * InternalTicker.NANOS_PER_TICK / InternalTicker.E6);
-        letPlayerSurvive();            // Make sure there are no obstacles
-        sleep(LONG);
-    }
-
     private void clickPopup(final int buttonNo) {
         if (AbstractScene.getPopup() != null) {
```
Comment (thervh70) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: This line isn't modified, but a test generally should not contain an if-statement. This branch is perfect to refactor this: in my opinion you should either remove this or add an else statement with a failure.

< testing, review

File: src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java (hunk 2 above)
Comment (mpsijm) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: This constant is used in #174c. I just left it here.

<review, layout

File: src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java (hunk 3 above)
Comment (thervh70) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: Fixed by @mpsijm. He added else statement.

**Comment by thervh70** [comment]
Thanks for fixing the things i've pointed out.
The amount of test line coverage is now 90% again!
This is ready to be merged.

< review

File: src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java (hunk 3 above)
Comment (mdingjan) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: Ye, but it causes a PMD warning.. :P I will fix it, that will be easier. ;)

Comment (mpsijm) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: Thanks for fixing, I overlooked the warning :)

<review

**Comment by mdingjan** [comment]
I reviewed your changes and fixed the last PMD warning. Ready for merging now. 
Will merge!

<review

**Pull request merged**